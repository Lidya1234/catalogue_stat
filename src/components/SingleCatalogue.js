import { PropTypes } from 'prop-types';

const SingleCatalogue = ({
  item, name, description,
}) => (
  <div className="single-catalog card" data-testid="catalogue">
    <h1 className="item-name">{item}</h1>
    <img src={name} alt={name} />
    <h4><i>{description}</i></h4>
  </div>
);

SingleCatalogue.propTypes = {
  item: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};
export default SingleCatalogue;
