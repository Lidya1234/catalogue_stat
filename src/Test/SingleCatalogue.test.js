import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import SingleCatalogue from '../components/SingleCatalogue';

describe('The singleCatalogue component should render with this information', () => {
  const category = {
 
    strCategory: 'Beef',
    strCategoryDescription: 'Beef is the culinary name for meat from cattle, particularly skeletal muscle. Humans have been eating beef since prehistoric times.[1] Beef is a source of high-quality protein and essential nutrients.[2]',
    strCategoryThumb: 'https://www.themealdb.com/images/category/beef.png',
  };
  render(
    <Router>
      <SingleCatalogue

        item={category.strCategory}
        name={category.strCategoryThumb}
        description={category.strCategoryDescription}
      />
    </Router>,
  );

  const component = renderer.create(
    <Router>
      <SingleCatalogue
        key={category.idCategory}
        item={category.strCategory}
        name={category.strCategoryThumb}
        description={category.strCategoryDescription}
      />
    </Router>,
  ).toJSON();
  const catalogue = screen.getByTestId('catalogue');
  test('Component should Render', () => {
    expect(catalogue).toBeInTheDocument();
  });
  test('Matchs snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
