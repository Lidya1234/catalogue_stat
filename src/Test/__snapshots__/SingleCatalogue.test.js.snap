// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Footer should match with snapshot Matchs snapshot 1`] = `
<div
  className="single-catalog card"
  data-testid="catalogue"
>
  <h1
    className="item-name"
  >
    Beef
  </h1>
  <img
    alt="https://www.themealdb.com/images/category/beef.png"
    src="https://www.themealdb.com/images/category/beef.png"
  />
  <h5>
    <i>
      Beef is the culinary name for meat from cattle, particularly skeletal muscle. Humans have been eating beef since prehistoric times.[1] Beef is a source of high-quality protein and essential nutrients.[2]
    </i>
  </h5>
</div>
`;

exports[`The single Catalogue component should render with this information Matchs snapshot 1`] = `
<div
  className="single-catalog card"
  data-testid="catalogue"
>
  <h1
    className="item-name"
  >
    Beef
  </h1>
  <img
    alt="https://www.themealdb.com/images/category/beef.png"
    src="https://www.themealdb.com/images/category/beef.png"
  />
  <h5>
    <i>
      Beef is the culinary name for meat from cattle, particularly skeletal muscle. Humans have been eating beef since prehistoric times.[1] Beef is a source of high-quality protein and essential nutrients.[2]
    </i>
  </h5>
</div>
`;

exports[`The singleCatalogue component should render with this information Matchs snapshot 1`] = `
<div
  className="single-catalog card"
  data-testid="catalogue"
>
  <h1
    className="item-name"
  >
    Beef
  </h1>
  <img
    alt="https://www.themealdb.com/images/category/beef.png"
    src="https://www.themealdb.com/images/category/beef.png"
  />
  <h5>
    <i>
      Beef is the culinary name for meat from cattle, particularly skeletal muscle. Humans have been eating beef since prehistoric times.[1] Beef is a source of high-quality protein and essential nutrients.[2]
    </i>
  </h5>
</div>
`;
